samples/bluetooth/fast_pair/locator_tag/src/fp_adv.c

```diff
@@ -347,7 +342,9 @@ static int fp_adv_set_teardown(void)
 
 static void fp_adv_restart_work_handle(struct k_work *w)
 {
-	fp_advertising_update();
+	if (app_fp_adv_is_ready()) {
+		fp_advertising_update();
+	}
```

Comment:
We need to react to them so we can properly clean the `fp_conn` pointer in the disconnected callback during the `app_fp_adv_disable` operation (in case the disable operation happens with the pending connection).

Alternatively, we could use a semaphore for synchronization but this introduces additional assumptions about the calling context of the `app_fp_adv_disable` (you cannot use the System workqueue which is used by the Bluetooth `disconnected` callback as it creates a deadlock).

Reply:
Oh, I understand. I'm resolving it then. Thanks!